drivers/interrupt_controller/intc_plic.c

```diff
@@ -146,54 +187,58 @@ void riscv_plic_set_priority(uint32_t irq, uint32_t priority)
  * This routine returns the RISCV PLIC-specific interrupt line causing an
  * interrupt.
  *
+ * @param dev Optional device pointer to get the interrupt line's controller
+ *
  * @return PLIC-specific interrupt line causing an interrupt.
  */
-int riscv_plic_get_irq(void)
+unsigned int riscv_plic_get_irq(const struct device *dev)
```

Comment:
It would be better to add a new function called something like `uint32_t riscv_plic_get_irq_by_dev(const struct device *dev)` instead of modifying the signature of the existing code.

With that, `unsigned int riscv_plic_get_irq(void)` can become a 1-line wrapper around the new function (passing `NULL`) and fewer 1-line changes are needed in e.g. unrelated gpio drivers

Reply:
fixed